webapp/sources/rudder/rudder-rest/src/test/scala/com/normation/rudder/MockServices.scala

```diff
@@ -677,3 +697,164 @@ class MockCompliance(mockDirectives: MockDirectives) {
+
+    override def closeAllOpenSession(endDate: DateTime, endCause: String): IOResult[Unit] = ???
+
+    override def getLastPreviousLogin(userId: String, closedSessionsOnly: Boolean): IOResult[Option[UserSession]] = {
+      userSessions.find(_.userId == userId).succeed
+    }
+
+    override def deleteOldSessions(olderThan: DateTime): IOResult[Unit] = ???
+
+    override def setExistingUsers(origin: String, users: List[String], trace: EventTrace): IOResult[Unit] = ???
+
+    override def disable(
+        userId:            List[String],
+        notLoggedSince:    Option[DateTime],
+        excludeFromOrigin: List[String],
+        trace:             EventTrace
+    ): IOResult[List[String]] = {
+      userId.succeed
+    }
+
+    override def delete(
+        userId:            List[String],
+        notLoggedSince:    Option[DateTime],
+        excludeFromOrigin: List[String],
+        trace:             EventTrace
+    ): IOResult[List[String]] = {
+      userId.succeed
+    }
+
+    override def purge(
+        userId:            List[String],
+        deletedSince:      Option[DateTime],
+        excludeFromOrigin: List[String],
+        trace:             EventTrace
+    ): IOResult[List[String]] = ???
+
+    override def setActive(userId: List[String], trace: EventTrace): IOResult[Unit] = {
+      ZIO.unit
+    }
+
+    override def updateInfo(
+        id:        String,
+        name:      Option[Option[String]],
+        email:     Option[Option[String]],
+        otherInfo: Option[Json.Obj]
+    ): IOResult[Unit] = {
+      ZIO.unit
+    }
+
+    override def getAll(): IOResult[List[UserInfo]] = userInfos.succeed
+
+    override def get(userId: String): IOResult[Option[UserInfo]] = {
+      userInfos.find(_.id == userId).succeed
+    }
+
+  }
+
+  val usersInputStream: () => InputStream = () => IOUtils.toInputStream(usersFile.contentAsString, StandardCharsets.UTF_8)
+
+  val userService: FileUserDetailListProvider = {
+    val usersFile = UserFile("test-users.xml", usersInputStream)
+
+    val roleApiMapping = new RoleApiMapping(AuthorizationApiMapping.Core)
+
+    val userFileProcessing = new UserFileProcessing(0)
+
+    val res = new FileUserDetailListProvider(roleApiMapping, usersFile, userFileProcessing)
+    res.reload()
+    res
+  }
+
+  val userManagementService: UserManagementService =
+    new UserManagementService(userRepo, userService, UserFile(usersFile.pathAsString, usersInputStream).succeed)
+
+  val providerRoleExtension: Map[String, ProviderRoleExtension] = Map("file" -> ProviderRoleExtension.WithOverride)
+  val authBackendProviders:  Set[String]                        = Set("file")
+}
+
+object MockUserManagement {
+  // Default mock with fake values and returning the temporary directory for cleanup
+  def apply(): (File, MockUserManagement) = {
+    val tmpDir = File.newTemporaryDirectory("rudder-users")
+    (tmpDir, new MockUserManagement(fakeUsers, fakeUserSessions, fakeUserFile(tmpDir)))
+  }
```

Comment:
Each time we instantiate a `MockUserManagement` we will need to cleanup the temp directory